### ISSUE TITLE
Lint for cast to inline class

### ISSUE THREAD

**Issue by Cat-sushi** [dart-lang/linter#4515]
I understand that `as` is a responsibility delegation from the compiler to users.
And, the compiler doesn't check down casts with `as`, but the runtime does.

On the other hand, the runtime skips user-defined check logic on constructor of `inline class`.

So, I propose a new lint to warn a cast from the representation type to the `inline class` with check logic or body on its constructor, without implicit constructor.

**Comment by srawlins** [comment]
CC @pq I don't think this has been implemented yet...

**Comment by pq** [comment]
Correct. No yet implemented. (But a really interesting idea!)

**Comment by eernstg** [comment]
@Cat-sushi, the targeted feature has been renamed from 'inline class' to 'extension type'. I adjusted the title accordingly. Please re-edit as needed if you disagree.

**Comment by srawlins** [comment]
@matanlurey and I were discussing: it would be good if this caught some non-explicit as-casts that stem from SDK libraries as well:

```dart
void f(List<int> employeeIds) {
  employeeIds.cast<EmployeeId>();
}
```

should be reported. And other cast methods from the SDK.
